### PR TITLE
Checkstyle

### PR DESCRIPTION
**Pull request by tisonkun**

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,438 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (yiheng) [inline checkstyle.xml]: What's the relationship of the file with this [one](https://github.com/Tencent/TubeMQ/blob/master/codestyle/intellij-java-google-style.xml)?

File: checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+        </module>
+
+        <module name="UnusedImports">
+            <property name="severity" value="error"/>
+            <property name="processJavadoc" value="true"/>
+            <message key="import.unused"
+                     value="Unused import: {0}."/>
+        </module>
+
+<!--        &lt;!&ndash;-->
```
Comment (yiheng) [inline checkstyle.xml]: can we remove the comment part?

File: checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+        </module>
+
+        <module name="TypeNameCheck">
+            <!-- Validates static, final fields against the
+            expression "^[A-Z][a-zA-Z0-9]*$". -->
+            <metadata name="altname" value="TypeName"/>
+            <property name="severity" value="error"/>
+        </module>
+
+<!--        <module name="ConstantNameCheck">-->
```
Comment (yiheng) [inline checkstyle.xml]: can we remove the comment part?

Comment (yiheng) [reply checkstyle.xml]: same with the following

File: checkstyle.xml [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+
+        <module name="ParenPad">
+            <!-- Checks that there is no whitespace before close parens or after
+                 open parens.
+            -->
+            <property name="severity" value="error"/>
+        </module>
+
+    </module>
+</module>
```
Comment (yiheng) [inline checkstyle.xml]: Tubemq should have some GUI code. Is it possible to include the code style check for the GUI code?

File: checkstyle.xml [hunk 5]
```diff
@@ -0,0 +1,436 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (guangxuCheng) [inline checkstyle.xml]:
```
 * Tencent is pleased to support the open source community by making TubeMQ available.
 *
 * Copyright (C) 2012-2019 Tencent. All Rights Reserved.
 *
```
Add the above content to the license

File: suppressions.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0"?>
+<!--
```
Comment (guangxuCheng) [inline suppressions.xml]:
```
 * Tencent is pleased to support the open source community by making TubeMQ available.
 *
 * Copyright (C) 2012-2019 Tencent. All Rights Reserved.
 *
```
Add the above content to the license

File: checkstyle.xml (hunk 1 above)
Comment (tisonkun) [reply checkstyle.xml]: Removed

File: checkstyle.xml (hunk 2 above)
Comment (tisonkun) [reply checkstyle.xml]: Information delivered.

File: checkstyle.xml (hunk 3 above)
Comment (tisonkun) [reply checkstyle.xml]: Information delivered.

File: checkstyle.xml (hunk 4 above)
Comment (tisonkun) [reply checkstyle.xml]: checkstyle is mainly aimed at checking Java code. For GUI I guess you refer to JS codes. Maybe their developer knows how to do that codestyle check.

**Comment by tisonkun** [comment]
Thanks for your reviews @yiheng @guangxuCheng . I narrow the pending issues to three below.

1. `ConstantNameCheck`.

Generally disabled on `private static final` so that we narrow the impact to `logger`. All `Logger` are now `logger` and we can always do a reformat if further concern occurs. The problem is that I modify several `logger` from `static final` to `private static final` but I think it is OK because all of them is not and should not be used outside the class.

2. `MethodNameCheck`

I have no idea whether changes to method name would impact existing dependencies to TubeMQ. If we all agree that we can rename `Subscribe` as `subscribe` and `TestXXX`/`test_xxx` as `testXxx` I will turn on the checker and do a general reformatting.

3. `RedundantModifier`

Mostly it is about `final` in methods of interfaces. An ideal way in my mind is reformatted our method signature as the format below

```java
<return type> methodName(
    T1 arg1,
    T2 arg2,
     ....
    TN arsN
) throws E1, E2 {
  ...
}
```

But it possibly impacts too much. An alternative is that we exclude `METHOD_DEF` property. It also makes sense to me.

CC @JunpingDu @gosonzhang

**Comment by yiheng** [comment]
+1

**Comment by tisonkun** [comment]
Thanks for your review @yiheng . Formally we send an approved review feedback for an approval.

Waiting for another reviewer @guangxuCheng since this pull request touch a lot of files.

**Comment by guangxuCheng** [comment]
@TisonKun  I think we should add a checker for the licence to ensure that the header of the file contains the licence.

**Comment by tisonkun** [comment]
@guangxuCheng good point! Implemented by apache-rat-plugin https://github.com/Tencent/TubeMQ/pull/55/commits/2ea61bf768b1df80cc59cd2f73194d65b0ae19c0

**Comment by guangxuCheng** [comment]
LGTM +1

**Pull request merged**